### PR TITLE
[IE CLDNN] Fully connected MMAD kernel simd16 improvements

### PR DESCRIPTION
**Pull request by lznamens**
Fully connected MMAD kernel improvements related to simd16 implementation. The result is several small improvements for TGL (more details are in JIRA ticket).

JIRA: 43945

### CODE REVIEW

File: inference-engine/thirdparty/clDNN/api/tensor.hpp [hunk 1]
```diff
@@ -273,8 +275,10 @@ struct format {
                 { image_2d_weights_c1_b_fyx,                   { 1, 1, 2, 0, 0, "oiyx",   "oixy?",      {}}},
                 { lstm_weights_dio,                            { 1, 1, 2, 0, 0, "oixy",   "oixy?",      {}}},
                 { os_is_yx_isa8_osv8_isv4,                     { 1, 1, 2, 0, 0, "oiyx",   "oixy?",      {}}},
+                { os_is_yx_isa8_osv16_isv4,                    { 1, 1, 2, 0, 0, "bfyx",   "bfxy?",      {}}},
                 { os_is_yx_isa8_osv8_isv4_swizzled_by_4,       { 1, 1, 2, 0, 0, "oiyx",   "oixy?",      {}}},
                 { os_is_zyx_isa8_osv8_isv4,                    { 1, 1, 3, 0, 0, "oizyx",  "oixyz",      {{1, 8}, {0, 8}, {1, 4}}}},
+                { os_is_zyx_isa8_osv16_isv4,                   { 1, 1, 3, 0, 0, "bfzyx",  "bfxyz",      {{1, 8}, {0, 16}, {1, 4}}}},
```
Comment (sshlyapn) [inline inference-engine/thirdparty/clDNN/api/tensor.hpp:281]: Please use `o` and `i` chars for weights formats

File: inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/include/fetch.cl [hunk 2]
```diff
@@ -715,6 +715,63 @@ inline uint FUNC(get_os_is_zyx_isa8_osv8_isv4_index)(uint o, uint i, uint z, uin
...
+    FUNC_CALL(get_os_is_yx_isa8_osv16_isv4_index)(                    \
+        o, i, y, x, CAT(prefix, _SIZE_X ),                            \
+        CAT(prefix, _SIZE_Y),                                         \
+        CAT(prefix, _IFM_NUM),                                        \
+        CAT(prefix, _OFM_NUM),                                        \
+        CAT(prefix, _OFFSET))
+
+inline uint FUNC(get_os_is_zyx_isa8_osv16_isv4_index)(uint o, uint i, uint z, uint y, uint x,
+                                                     uint size_x, uint size_y, uint size_z,
+                                                     uint size_ifm, uint size_ofm, uint offset)
```
Comment (sshlyapn) [inline inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/include/fetch.cl:746]: nit: please fix alignment

**Review by sshlyapn — APPROVED** [review]
In general looks good to me

File: inference-engine/thirdparty/clDNN/api/tensor.hpp (hunk 1 above)
Comment (lznamens) [reply inference-engine/thirdparty/clDNN/api/tensor.hpp]: Thanks. Fixed.

File: inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/include/fetch.cl (hunk 2 above)
Comment (lznamens) [reply inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/include/fetch.cl]: Fixed.

**Pull request merged**